dashboard/app/controllers/aichat_controller.rb

```diff
@@ -15,24 +15,46 @@ def chat_completion
     end
 
     # Check for PII / Profanity
-    # Copied from ai_tutor_interactions_controller.rb - not sure if filtering is working.
     locale = params[:locale] || "en"
-    new_message_text = params[:newMessage]
-    # Check only the newest message from the user for inappropriate content.
-    filter_result = ShareFiltering.find_failure(new_message_text, locale) if new_message_text
-    # If the content is inappropriate, we skip sending to endpoint and instead hardcode a warning response on the front-end.
-    return render(status: :ok, json: {status: filter_result.type, flagged_content: filter_result.content}) if filter_result
-
-    input_json = AichatSagemakerHelper.format_inputs_for_sagemaker_request(params[:aichatModelCustomizations], params[:storedMessages], params[:newMessage])
-    sagemaker_response = AichatSagemakerHelper.request_sagemaker_chat_completion(input_json)
+    filter_result = ShareFiltering.find_failure(params[:newMessage], locale)
+    if filter_result
+      new_messages = [
+        {
+          role: 'user',
+          content: params[:newMessage],
+          status: SharedConstants::AICHAT_INTERACTION_STATUS[:PROFANITY_VIOLATION]
+        }
+      ]
+      session_id = log_chat_session(new_messages)
+      return render(
+        status: :ok,
+        json: {
+          status: filter_result.type,
+          flagged_content: filter_result.content,
+          session_id: session_id
+        }
+      )
+    end
+
+    # Use to_unsafe_h here to allow testing this function.
+    # Safe params are primarily targeted at preventing "mass assignment vulnerability"
+    # which isn't relevant here.
+    input = AichatSagemakerHelper.format_inputs_for_sagemaker_request(
+      params.to_unsafe_h[:aichatModelCustomizations],
+      params.to_unsafe_h[:storedMessages].filter {|message| message[:status] == SharedConstants::AICHAT_INTERACTION_STATUS[:OK]},
+      params.to_unsafe_h[:newMessage]
+    )
+    sagemaker_response = AichatSagemakerHelper.request_sagemaker_chat_completion(input)
     latest_assistant_response = AichatSagemakerHelper.get_sagemaker_assistant_response(sagemaker_response)
-    assistant_message = {
-      role: "assistant",
-      content: latest_assistant_response,
-    }
 
-    session_id = log_chat_session(assistant_message)
-    return render(status: :ok, json: assistant_message.merge({sessionId: session_id}).to_json)
+    assistant_message = {role: "assistant", content: latest_assistant_response, status: SharedConstants::AICHAT_INTERACTION_STATUS[:OK]}
```

Comment:
Just curious - you mentioned that follow-up would include checking response from Sagemaker. Are we planning to just use WebPurify as we do for user messages?

Reply:
Yes that's my understanding, does that sound right to you?